### PR TITLE
[stable/prometheus-operator] add ingress for Thanos sidecar, update kubelet svc

### PR DESCRIPTION
**Pull request by mpritter76**
#### What this PR does / why we need it:
* add ingress configuration for Thanos sidecar, so it can be accessed externally by a centralized thanos querier running in another cluster
* update kubelet service monitor to have scrape timeout, so we don't get time outs on slow kubelets
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by mpritter76** [comment]
@gianrubio, @vsliouniaev Any timing on when this might get looked at?

File: stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.kubelet.enabled }}
-apiVersion: monitoring.coreos.com/v1
+apiVersion: {{ printf "%s/v1" (.Values.prometheusOperator.crdApiGroup | default "monitoring.coreos.com") }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/exporters/kubelet/servicemonitor.yaml]: Support for changing the crdApiGroup was removed entirely in prometheus-operator, so supporting it is not necessary.

File: stable/prometheus-operator/templates/prometheus/ingressThanosSidecar.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{{- $routePrefix := list .Values.prometheus.prometheusSpec.routePrefix }}
+{{- $paths := .Values.prometheus.thanosIngress.paths | default $routePrefix -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+{{- if .Values.prometheus.thanosIngress.annotations }}
+  annotations:
+{{ toYaml .Values.prometheus.thanosIngress.annotations | indent 4 }}
+{{- end }}
+  name: thanos-gateway-{{ .Release.Name }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/prometheus/ingressThanosSidecar.yaml]: This doesn't seem to be following naming conventions used elsewhere.

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/templates/prometheus/ingressThanosSidecar.yaml (hunk 2 above)
Comment (mpritter76) [reply stable/prometheus-operator/templates/prometheus/ingressThanosSidecar.yaml]: Would this be more preferred? 
```
name: {{ template "prometheus-operator.fullname" . }}-thanos-gateway
```

**Comment by mpritter76** [comment]
@vsliouniaev  Its not really clear why the test is failing!

**Comment by vsliouniaev** [comment]
You need to add signoff on all commits to submit valid PRs to this repo. [The DCO check explains how to fix this](https://github.com/helm/charts/pull/22806/checks?check_run_id=858982427)

**Comment by mpritter76** [comment]
Closing ... opened a new PR https://github.com/helm/charts/pull/23200

**Pull request closed without merging**